packages/ckeditor5-dev-env/lib/release-tools/tasks/bump-year.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const fs = require( 'fs' );
+const glob = require( 'glob' );
+const minimatch = require( 'minimatch' );
+
+/**
+ * See https://github.com/ckeditor/ckeditor5/issues/10909
+ *
+ * @param {Object} params
+ * @param {String} params.cwd Current working directory from which all paths will be resolved
+ * @param {Function} params.callback Optional function to modify the license header
+ * @param {Number} params.initialYear Year from which the licenses should begin
+ * @param {Array} params.globPatterns array of objects, where each object has string property 'pattern',
+ * and optionally 'options' property.
+ */
+module.exports = function bumpYear( params ) {
+	params.globPatterns
+		.map( globPattern => glob.sync( globPattern.pattern, globPattern.options ) )
+		.reduce( ( previous, current ) => [ ...previous, ...current ] )
+		.filter( fileName => {
+			// Filter out nested `node_modules`.
+			if ( minimatch( fileName, '**/node_modules/**' ) ) {
+				return false;
+			}
+
+			// Filter out stuff from `src/lib/`.
+			if ( minimatch( fileName, '**/src/lib/**' ) ) {
+				return false;
+			}
+
+			// Filter out builds.
+			if ( minimatch( fileName, '**/ckeditor5-build-*/**' ) ||
+				minimatch( fileName, '**/build/**' ) ) {
+				return false;
+			}
+
+			// Filter out directories.
+			if ( fs.statSync( fileName ).isDirectory() ) {
+				return false;
+			}
+
+			return true;
+		} )
```

Comment:
These rules should not be part of the main script. This script should only take the glob patterns and find files that matches them.

Reply:
The `glob` package has the [`ignore` option](https://github.com/isaacs/node-glob#options) - maybe it could be used to do the filtering on the low-level?

Reply:
I agree that we could remove `src/lib/` from there. However:

*   filtering out directories is obvious
*   it would probably make sense to filter out `node_modules` everywhere
*   while not all projects might have `build` directory, we probably want to filter out all that do anyway

The last point is debatable, however, at least the first two surely should be kept in there?

Reply:
After rethinking it, you are right. The `**/node_modules/**` and `**/build/**` directories should be always skipped. Also filtering out the directories here makes sense, as we want to target only files. But the `**/src/lib/**` and `**/ckeditor5-build-*/**` patterns are strictly related to the CKEditor 5 repository so they should not be here.

My idea was to speed up this script a bit, so that it doesn't have to go to the deeply nested `node_modules` and other directories (i.e. `**/build/**`), which we'll filter out anyway. To accomplish this, it looks like it is enough to provide the `ignore` option (an array of patterns) to the `glob` that should not be traversed. As soon as `glob` finds a path that matches the ignored pattern, it does not go deeper. Would you like to investigate a bit to check if this is doable? If yes, then 🤞🏻. If not, then please just extract the two `**/src/lib/**` and `**/ckeditor5-build-*/**` patterns out from this file, as they are related to the CKEditor 5 only.

Reply:
```
pattern: '!(build|coverage|node_modules|external)/**',
options: {
    ignore: [
        '**/ckeditor5-build-*/**',
        '**/src/lib/**'
    ]
}
```

☝️ this works. Funny thing is, that current version on master has this line:

[https://github.com/ckeditor/ckeditor5/blob/84b676ee99137617418498b54494852ffbfcbba9/scripts/bump-year.js#L49](https://github.com/ckeditor/ckeditor5/blob/84b676ee99137617418498b54494852ffbfcbba9/scripts/bump-year.js#L49)

But it does not work, and it updates `**/ckeditor5-build-*/**` anyway

Reply:
In my previous comment my point was to get rid of the `.filter()` method at all and provide the common ignore patterns as the default option to the `glob`. Then, `glob` will not need to go inside the deeply nested directories that we are skipping anyway. The common options for glob could be something like:

```
{
  nodir: true,
  ignore: [
    '**/node_modules/**'
  ]
}
```

These options should be then merged with the options received as an arguments to this script.